### PR TITLE
REGR: get_loc for ExtensionEngine not returning bool indexer for na

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Stumbled across this when working on something else.

ExtensionEngine was new in 1.5, so no whatsnew. But should backport to 1.5

### CODE REVIEW

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @phofl

**Review by mroeschke — APPROVED** [review]
LGTM, could you merge in main once more? (Checking that the failures are definitely not related)

File: pandas/tests/indexes/test_indexing.py [hunk 1]
```diff
@@ -221,6 +222,13 @@ def test_get_loc_generator(self, index):
             # MultiIndex specifically checks for generator; others for scalar
             index.get_loc(x for x in range(5))
 
+    def test_get_loc_masked_duplicated_na(self):
+        # GH#48411
+        idx = Index([1, 2, NA, NA], dtype="Int64")
+        result = idx.get_loc(NA)
+        expected = np.array([False, False, True, True])
```
[the remark this answers is not in the document]
Comment (phofl) [reply pandas/tests/indexes/test_indexing.py:229]: Oh correct, but only if no NAs are present at all. Have to check in a follow up what we want to do here.

**Comment by phofl** [comment]
One of the failures was related, int32 instead of int64 dtype. Fixed this since it is not really relevant here

File: pandas/tests/indexes/test_indexing.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (phofl) [reply pandas/tests/indexes/test_indexing.py:229]: Actually this is correct. We are only returning slices when the Index is monotonic increasing.

```
idx2 = Index([1, 2, 2, 3, 3, 0], dtype="int64")
idx2.get_loc(2)
```

This returns a bool indexer too

[the remark this answers is not in the document]
Comment (jorisvandenbossche) [reply pandas/tests/indexes/test_indexing.py:229]: Ah, good point, it are the missing values that is making it _not_ monotonic, I suppose. 
I tested it with replacing the NAs with a number, in which case it _was_ monotonic:

```
In [8]: idx = Index([1, 2, 3, 3], dtype="Int64")

In [9]: idx.get_loc(3)
Out[9]: slice(2, 4, None)
```

[the remark this answers is not in the document]
Comment (phofl) [reply pandas/tests/indexes/test_indexing.py:229]: Yep correct. Adding NA makes it non-monotonic in all cases.

**Pull request merged**